include/dma.h

```diff
@@ -197,7 +197,9 @@ static inline int dma_config(struct device *dev, u32_t channel,
  * @retval 0 if successful.
  * @retval Negative errno code if failure.
  */
-static inline int dma_start(struct device *dev, u32_t channel)
+__syscall int dma_start(struct device *dev, u32_t channel);
```

Comment:
Do we need DMA for user threads?  Because if we do, then it's something we need to think very carefully, as this can be used to access arbitrary memory locations.

Reply:
I don't have a clear sense of what subsystems are needed for user threads.

However keep in mind that subsystem APIs cannot be used on a driver object unless that thread has been granted permission, and the dma_config() API was omitted.

However if you think it's best to just omit DMA from user mode access (or any other subsystem) I can certainly remove it.